src/framework/components/layout-group/system.js

```diff
@@ -48,40 +48,13 @@ class LayoutGroupComponentSystem extends ComponentSystem {
         if (data.reverseX !== undefined) component.reverseX = data.reverseX;
         if (data.reverseY !== undefined) component.reverseY = data.reverseY;
         if (data.alignment !== undefined) {
-            if (data.alignment instanceof Vec2) {
-                component.alignment.copy(data.alignment);
-            } else {
-                component.alignment.set(data.alignment[0], data.alignment[1]);
-            }
-
-            /* eslint-disable no-self-assign */
-            // force update
-            component.alignment = component.alignment;
-            /* eslint-enable no-self-assign */
+            component.alignment = Array.isArray(data.alignment) ? new Vec2(data.alignment) : data.alignment;
```

Comment:
looks good, but should it maybe create new Vec2 in both cases (even when it already is Vec2), to avoid directly referencing data.alignment? Or does it get copied in .alignment?

Reply:
It gets copied in the setter, so it's fine. It's how all component properties work actually.

Reply:
I take that back. The component takes a reference. Wow, that's wrong. I'll need to consider this a bit further.

Reply:
OK, I've updated the component to take copies of vectors instead of storing a reference to what's set. I think the PR's good now.